ratelimits/bucket.go

```diff
@@ -0,0 +1,56 @@
+package ratelimits
+
+import (
+	"fmt"
+	"net"
+)
+
+// Bucket identifies a specific subscriber rate limit bucket to the Limiter.
+type Bucket struct {
+	name Name
+	key  string
+}
+
+// BucketWithCost is a bucket with an associated cost.
+type BucketWithCost struct {
+	Bucket
+	cost int64
```

Comment:
How do you feel about moving the `cost` into the bucket itself rather than having a `BucketWithCost` which embeds a `Bucket`? All of the dramatic `cost` changes appear to be in testing, and the WFE is setting a `cost` of 1.

You could remove the `WithCost` method and the constructors below could set a default `cost: 1,` which can be overridden in each `//ratelimits/limiter_test` as ` bucket.cost = 20` etc.

Reply:
I honestly don't like losing the composability in testing.

Reply:
Additionally, we absolutely want to keep the variable cost flexibility offered by the GCRA algorithm.

Say we have `certsPerDomain` limit to check for three eTLD+1 in the cert:
- example.com
- example.net
- example.rofl

This fails because we've issued for example.rofl too many times recently. However the subscriber has an override for their ACME account. So we need to attribute `certsPerDomainPerRegID` with cost of 3 as part of the batch. Because of cost variability, we can GET their bucket, make 1 decision with a cost of 3 and (if allowed) we can PUT with a cost of 3.

Reply:
That makes a lot of sense.